enumerables.rb

```diff
@@ -0,0 +1,168 @@
+module Enumerable
+  # my_each
+  def my_eac
+    return enum_for(:my_each) unless block_given?
+
+    arr = self if self.class == Array
+    arr = to_a if self.class == Range || Hash
+    0.upto(arr.length - 1) do |index|
+      yield(arr[index])
+    end
+    arr
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    arr = self if self.class == Array
+    arr = to_a if self.class == Range || Hash
+
+    0.upto(arr.size - 1) do |index|
+      yield(arr[index], index)
+    end
+    arr
+  end
+
+  # my_select
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    new_a = []
+    my_each do |element|
+      new_a << element if yield(element)
+    end
+    new_a
+  end
+
+  # My_all?
+  def my_all?(parameter = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+      return true
+    elsif parameter.nil?
+      my_each { |n| return false if n.nil? || n == false }
+    elsif !parameter.nil? && (parameter.is_a? Class)
+      my_each { |n| return false unless [n.class, n.class.superclass].include?(parameter) }
+    elsif !parameter.nil? && patameter.class == Regexp
+      my_each { |n| return false unless parameter.match(n) }
+    else
+      my_each { |n| return false if n != parameter }
+    end
+    true
+  end
```

Comment:
- my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
- my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
- my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern